include/envoy/runtime/runtime.h

```diff
@@ -112,12 +125,12 @@ class Snapshot {
    *        does not exist or it is not an integer.
    * @param random_value supplies the stable random value to use for determining whether the feature
    *        is enabled.
-   * @param control max number of buckets for sampling. Sampled value will be in a range of
-   *        [0, num_buckets).
+   * @param num_buckets control max number of buckets for sampling. Sampled value will be in a range
+   *        of [0, num_buckets).
    * @return true if the feature is enabled.
    */
-  virtual bool featureEnabled(const std::string& key, uint64_t default_value, uint64_t random_value,
-                              uint64_t num_buckets) const PURE;
+  virtual bool sampleFeatureEnabled(const std::string& key, uint64_t default_value,
+                                    uint64_t random_value, uint64_t num_buckets) const PURE;
```

Comment:
@mattklein123 how are you guys doing fractional percent with runtimes? IOW, do you need runtime support for fractional percent? If not, all changes to this file can be eliminated

Reply:
I think `access_log_impl` uses `runtime` for sampling the feature status as seen [here](https://github.com/envoyproxy/envoy/blob/2a68458119ffcd97498fcbe13fc14c8b8d043cb0/source/common/access_log/access_log_impl.cc#L121-L123). Should we just stick to a similar approach for `fault`s?

Reply:
Looking again, why is this being renamed ?

Reply:
Because of a signature conflict between the routines highlighted [here](
https://github.com/venilnoronha/envoy/blob/f1056616cfb0b87e1b0be75102f89e452955a664/include/envoy/runtime/runtime.h).